x-pack/plugins/infra/public/hooks/use_lens_attributes.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect, useMemo, useState } from 'react';
+import type { DataView } from '@kbn/data-views-plugin/public';
+import { Filter, Query, TimeRange } from '@kbn/es-query';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import { FormulaPublicApi } from '@kbn/lens-plugin/public';
+import { ActionExecutionContext } from '@kbn/ui-actions-plugin/public';
+import { i18n } from '@kbn/i18n';
+import { InfraClientSetupDeps, LensAttributes, LensOptions } from '../types';
+import {
+  buildLensAttributes,
+  HostLensAttributesTypes,
+  hostMetricsLensAttributes,
+} from '../common/visualizations';
+
+interface UseLensAttributesParams {
+  type: HostLensAttributesTypes;
+  dataView: DataView | undefined;
+  options?: LensOptions;
+}
+
+export const useLensAttributes = ({
+  type,
+  dataView,
+  options = {
+    breakdownSize: 10,
+  },
+}: UseLensAttributesParams) => {
+  const {
+    services: { lens },
+  } = useKibana<InfraClientSetupDeps>();
+  const { navigateToPrefilledEditor } = lens;
+  const [formulaAPI, setFormulaAPI] = useState<FormulaPublicApi>();
+
+  useEffect(() => {
+    const getFormulaApi = async () => {
+      const { formula } = await lens.stateHelperApi();
+      return setFormulaAPI(formula);
+    };
+    getFormulaApi();
```

Comment:
**nit:** it results in the same result, but it uses less syntactic sugar and native promise implementation:
```js
lens.stateHelperApi().then(({ formula }) => setFormulaAPI(formula))
```
Also, is there any chance the `lens.stateHelperApi()` can fail and we need to handle this error with a catch?

Reply:
I'll think about how to handle errors